scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/reconciler/ReconcilerHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.scheduling.reconciler;
+
+import com.amazonaws.blox.dataservicemodel.v1.client.DataService;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.ListClustersRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.operations.ListClustersResponse;
+import com.amazonaws.blox.lambda.LambdaFunction;
+import com.amazonaws.blox.scheduling.manager.ManagerInput;
+import com.amazonaws.blox.scheduling.manager.ManagerOutput;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.spotify.futures.CompletableFutures;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+@Log4j2
+public class ReconcilerHandler implements RequestHandler<CloudWatchEvent<Map>, Void> {
+
+  final DataService dataService;
+  final LambdaFunction<ManagerInput, ManagerOutput> stateFunction;
+
+  @Override
+  public Void handleRequest(CloudWatchEvent<Map> input, Context context) {
+    ListClustersResponse r = dataService.listClusters(new ListClustersRequest());
```

Comment:
how are we thinking about handling exceptions? logging? (everywhere there are service calls)

Reply:
I'd like to put the error handling/request logging in middleware, probably between this handler and the top-level JacksonRequestStreamHandler. That's probably next after the state client.

Reply:
Did you figure out how the checked exceptions from dependencies are being thrown here? Can we wrap it in an application specific exception here before logging in middleware?

Reply:
The answer is that they aren't. The `listClusters` API call does not declare any checked exceptions in this PR. Once that is added to the method signature, then we can indeed to some exception wrapping either in the middleware, or at this callsite.